compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/FirHelpers.kt

```diff
@@ -523,6 +520,7 @@ private val FirSimpleFunction.matchesToStringSignature: Boolean
 
 fun checkTypeMismatch(
     lValueOriginalType: ConeKotlinType,
+    assignment: FirVariableAssignment?,
     rValue: FirExpression,
     context: CheckerContext,
     source: FirSourceElement,
```

Comment:
The `source` parameter here is actually `rValue.source` so you can just remove it so it's not confusing which element's source it is, and just use `rValue.source` everywhere (or a local variable for `rValue.source`).